Content.Shared/Damage/DamageableSystem.cs

```diff
@@ -0,0 +1,313 @@
+                // which we default to support all). Add every single damage type to our component:
+                foreach (var type in _prototypeManager.EnumeratePrototypes<DamageTypePrototype>())
+                {
+                    component.DamagePerType.Add(type.ID, 0);
+                }
+
+                DamageChanged(uid, component, false);
+                return;
+            }
+
+            // A valid damage container ID was provided. Initialize damage dictionary, using the types from the damage
+            // container prototype
+            component.DamagePerType = new(damageContainerPrototype.SupportedTypes.Count);
+            foreach (var type in damageContainerPrototype.SupportedTypes)
+            {
+                component.DamagePerType.Add(type, 0);
+            }
+
+            // Then also add the supported damage groups
+            foreach (var groupID in damageContainerPrototype.SupportedGroups)
+            {
+                var group = _prototypeManager.Index<DamageGroupPrototype>(groupID);
+                foreach (var type in group.DamageTypes)
+                {
+                    component.DamagePerType.TryAdd(type, 0);
+                }
+            }
+
+            DamageChanged(uid, component, false);
+        }
+
+        /// <summary>
+        ///     If the damage in a DamageableComponent was changed, this function should be called.
+        /// </summary>
+        /// <remarks>
+        ///     This updates cached damage information, flags the component as dirty, and raises a damage changed event.
+        ///     The damage changed event is used by other systems, such as damage thresholds.
+        /// </remarks>
+        public void DamageChanged(EntityUid uid, DamageableComponent component, bool damageIncreased)
+        {
+            component.TotalDamage = component.DamagePerType.Values.Sum();
+            component.DamagePerGroup = GetDamagePerGroup(component);
+            component.Dirty();
+            RaiseLocalEvent(uid, new DamageChangedEvent(component, damageIncreased), false);
+        }
+
+        /// <summary>
+        ///     Applies damage to the component, using damage specified via a <see cref="DamageSpecifier"/> instance.
+        /// </summary>
+        /// <remarks>
+        ///     <see cref="DamageSpecifier"/> is effectively just a dictionary of damage types and damage values. This
+        ///     function just applies the container's resistances (unless otherwise specified) and then changes the
+        ///     stored damage data. Division of group damage into types is managed by <see cref="DamageSpecifier"/>.
+        /// </remarks>
+        /// <returns>
+        ///     Returns false if a no damage change occurred; true otherwise.
+        /// </returns>
+        public void TryChangeDamage(EntityUid uid, DamageableComponent component, TryChangeDamageEvent args)
```

Comment:
The event has been removed now and replaced with a public method that takes damage and an entity as an argument. Currently the system just handles the damage itself if the entity has a damageable component. 

I made it take an entity as an argument based on sloth's comment. Passing the entity is useful if future components/systems have their own damage handling functionality (e.g., body system). That way you dont need to refactor every single damage dealing/healing system, and only need to add a bit of code to damageable system so that it can pass it on to the appropriate system.